### PR TITLE
Lint timeseries panels too

### PR DESCRIPTION
**Pull request by wuzuf**
Support timeseries panels

### CODE REVIEW

File: lint/rule_panel_rate_interval.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func NewPanelRateIntervalRule() *PanelRuleFunc {
 			}
 
 			switch p.Type {
-			case "singlestat", "graph", "table":
```
Comment (tomwilkie) [inline lint/rule_panel_rate_interval.go:26]: Can we update this function to use `panelHasQueries`?

Comment (wuzuf) [reply lint/rule_panel_rate_interval.go:26]: Sure, I proposed a new version below.
I just wonder if it is better like I did with an early return or if you prefer to protect the `for _, target := range p.Targets {...}` with an `if panelHasQueries(p)`

File: lint/rule_panel_rate_interval.go [hunk 2]
```diff
@@ -14,24 +14,21 @@ func NewPanelRateIntervalRule() *PanelRuleFunc {
...
 		description: "Checks that each panel uses $__rate_interval.",
 		fn: func(d Dashboard, p Panel) Result {
-			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" {
+			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" || !panelHasQueries(p) {
 				// Missing template datasources is a separate rule.
 				return Result{
 					Severity: Success,
 					Message:  "OK",
 				}
 			}
```
Comment (tomwilkie) [inline lint/rule_panel_rate_interval.go:23]: In this situation a more verbose check would be preferred, so future editors don't miss it.

```suggestion
			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" {
				// Missing template datasources is a separate rule.
				return Result{
					Severity: Success,
					Message:  "OK",
				}
			}
			
			if !panelHasQueries(p) {
				// Don't lint certain types of panels.
				return Result{
					Severity: Success,
					Message:  "OK",
				}
			}
```

**Review by tomwilkie — APPROVED** [review]
LGTM with one nit! Thank you @wuzuf

**Review by tomwilkie — APPROVED** [review]

**Pull request merged**